File: Geonil/problem_solving/week1/BOJ10828.py
import sys
In = sys.stdin.readline


class Stack:
    def __init__(self):
        self.top = -1
        self.stack = []

    def is_empty(self):
        if self.top == -1:
            return True
        else:
            return False

    def push_stack(self, data):
        self.stack.append(data)
        self.top += 1

    def pop_stack(self):
        if self.is_empty():
            return -1
        else:
            data = self.stack[self.top]
            del self.stack[self.top]
            self.top -= 1
            return data

    def get_size(self):
        return len(self.stack)

    def get_top(self):
        if self.is_empty():
            return -1
        else:
            return self.stack[self.top]


def main():
    num = int(In())
    stk = Stack()
    for _ in range(num):
        cmd = In().rstrip().split()
        if cmd[0] == 'push':
            stk.push_stack(int(cmd[1]))
        elif cmd[0] == 'pop':
            rlt = stk.pop_stack()
            print(rlt)
        elif cmd[0] == 'size':
            print(stk.get_size())
        elif cmd[0] == 'empty':
            if stk.is_empty():
                print(1)
            else:
                print(0)
        elif cmd[0] == 'top':
            print(stk.get_top())


main()

File: Taewoo/problem_solving/~2021.02/week1/BOJ10828.cpp
#include<iostream>
#include<stack>
using namespace std;

int main() {
    stack<int> stk;
    int N; cin>>N;

    for(int i=0; i<N; i++) {
        string input; cin>>input;
        
        if(input == "push") {
            int num; cin>>num;
            stk.push(num);
        }
        else if(input == "pop"){
            if(stk.size()) {
                cout<<stk.top()<<'\n';
                stk.pop();
            }
            else cout<<-1<<'\n';
        }
        else if(input == "size") {
            cout<<stk.size()<<'\n';
        }
        else if(input == "empty") {
            cout<<stk.empty()<<'\n';
        }
        else if(input == "top") {
            if(stk.size()) cout<<stk.top()<<'\n';
            else cout<<-1<<'\n';
        }
    }
    return 0;
}